src/sage/topology/simplicial_complex.py

```diff
@@ -4551,6 +4557,7 @@ def set_immutable(self):
             False
         """
         self._is_immutable = True
+        self._bbn = None
```

Comment:
This shouldn't be done in `set_immutable`. You just cleared away all the hard computations for saying the simplicial complex is now immutable, but you didn't change it. Instead, it should be in the `add_*` and `delete_*` methods that actually change (in-place) the complex.

Reply:
Yeah, that makes much more sense. Sorry for misinterpreting.